$OpenBSD: patch-src_ui_act_c,v 1.2 2012/04/21 13:38:47 dcoppa Exp $

Strict pthread implementations (like OpenBSD's rthreads with
PTHREAD_MUTEX_TYPE_STRICT_NP set) don't allow unlocking a mutex
that is not locked, resulting in abort() being called. Always
aquiring the lock before unlocking it while skipping to the next
song fixes this.
(upstream git commit bbed23df60c4bfc067b4f7c03574cfe11cc8bb60)

--- src/ui_act.c.orig	Fri Apr 20 09:01:59 2012
+++ src/ui_act.c	Sat Apr 21 15:26:21 2012
@@ -50,6 +50,8 @@ static inline void BarUiDoSkipSong (struct audioPlayer
 	assert (player != NULL);
 
 	player->doQuit = 1;
+	/* unlocking an unlocked mutex is forbidden by some implementations */
+	pthread_mutex_trylock (&player->pauseMutex);
 	pthread_mutex_unlock (&player->pauseMutex);
 }
 
